app/controllers/admin/purchases_controller.rb

```diff
@@ -134,7 +134,7 @@ def undelete
 
   def update_giftee_email
     new_giftee_email = params[:giftee_email]
-    gift = Gift.find_by(gifter_purchase_id: params[:id])
+    gift = Gift.find_by(gifter_purchase_id: @purchase.id)
 
     if gift.present? && new_giftee_email != gift.giftee_email
```

Comment:
You might as well use the relationship between the two now that `@purchase` is available

```suggestion
    if (gift = @purchase.gift_given).present? && new_giftee_email != gift.giftee_email
```